support-frontend/app/services/pricing/PriceSummaryService.scala

```diff
@@ -12,22 +12,28 @@ import org.joda.time.Months
 
 import scala.math.BigDecimal.RoundingMode
 
-class PriceSummaryService(promotionService: PromotionService, catalogService: CatalogService)
-    extends TouchpointService {
+class PriceSummaryService(
+    promotionService: PromotionService,
+    defaultPromotionsService: DefaultPromotionsService,
```

Comment:
@johnduffell what are your thoughts about passing the `defaultPromotionsService` into this object as a dependency rather than just adding the defaults to the codes that are passed in to the function. 

I can see why it's been done it this way but it is a bit weird that it requires it for a single method call and the fact that it then exposes the `defaultPromotionsService.getPromoCodes` method for external code feels like a bit of a code smell?

Reply:
I see what you mean, and it did feel a bit wrong exposing `getPromoCodes`. It's used in one external place, for getting promo copy. But it seems preferable for `PriceSummaryService` to add the default promos, rather than every use of `getPrices` having to know about them

Reply:
see what you mean - we are effectively partially applying all of the methods in this service with the whole DefaultPromotionsService, most of which are not relevant.  So it does make unit testing and reuse more tricky.

Structuring code is something it is easy to go round in circles with - we would need to follow how the code actually depends on each other to decide. 

 personally I'd be tempted to split the PriceSummaryService in half if `getPrices` is called from a lot of places that the other methods are not.  But if most places need most of the methods, it gets hard to justify.  But I do generally tend towards very small interfaces with minimal dependencies, which has its own limitations.
One precendent ( naming from memory) we created the ZuoraGiftingService and the ZuoraService separately to avoid creating unnecessary dependencies.